src/app/components/application/personal-info/parent/parent-info/parent-info.component.ts

```diff
@@ -109,6 +115,16 @@ export class ParentInfoComponent implements OnInit, OnChanges {
   }
 
   stateSelected(event: StateCode): void {
-    this.parent.mailingAddress.stateProvince = event.name;
+    this.parent.mailingAddress.stateProvince = event.isoCode;
+    this.parentState = this.filteredStates.find(x => x.isoCode === this.parent?.mailingAddress?.stateProvince)?.name;
+  }
+
+  getState(isoCode:string): string {
+    if (isoCode) {
+      let thisState = this.filteredStates.find(x => x.isoCode === isoCode);
```

Comment:
Instead of filtered, this should maybe be using the full state list? Feels like maybe there is the potential for the filtered list to not be populated yet when this method is being called by the parent data load stuff?

Same for the student version of this method.

Reply:
I had an issue with this actually. If I used the full list, then it would select Calgerie (or some other country's state) instead of California based off the isoCode

Reply:
So, when I selected California, it would show california, but when I saved and came back the name was a different state starting with 'ca'